### PR TITLE
N+1 and Integration tests

### PR DESCRIPTION
**Pull request by maov19**
### Implementations

1. N+1 problem is solved when fetching all posts and their comments for a user.
 1.1 Screenshots

**Before**
before-n1

**After**
after-n1

2. Create integration tests for all of the views used in your project. 
> 2.1. Covered all user workflows
- User index page
- User show page
- User post index page
- Post show page

### CODE REVIEW

File: spec/features/posts_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    it 'should show likes counter' do
+      expect(page).to have_content(@first_post.likes_counter)
+    end
+
+    it 'should navigate to post show page when clicked on post' do
+      find('a', text: @first_post.text).click
+      expect(current_path).to eq(user_post_path(@first_post.author_id, @first_post))
+    end
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline spec/features/posts_index_spec.rb]: - [ ] kindly add a Tests that asserts you can see a section for pagination if there are more posts than fit on the view.

File: spec/features/posts_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  scenario 'I can see the username, profile picture, and number of posts for each user' do
+    expect(page).to have_content(@user1.name)
+    expect(page.html).to include(@user1.photo)
+    expect(page).to have_content(@user1.posts_counter)
+  end
+
+  scenario 'When I click on a user, I am redirected to their show page' do
+    click_on @user1.name
+    expect(current_path).to eq(user_path(@user1))
+  end
```
Comment (VICTOR-CHUKWU) [inline spec/features/posts_show_spec.rb]: - [ ] kindly add the following tests i.e `Tests that asserts you can see the post's title.`, `Tests that asserts you can see who wrote the post.`, `Tests that asserts you can see how many comments it has.`, `Tests that asserts you can see how many likes it has.`, `Tests that asserts you can see the post body.`, `* Tests that asserts you can see the username of each commenter. Tests that asserts you can see the comment each commentor left.`

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @maov19  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/posts_show_spec.rb (hunk 2 above)
Comment (maov19) [reply spec/features/posts_show_spec.rb]: Added!
image

File: spec/features/posts_index_spec.rb (hunk 1 above)
Comment (maov19) [reply spec/features/posts_index_spec.rb]: Added! Thanks for the clarification.

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @maov19 and @edubew, 

To highlight 👍🏽 
✔️ Tests are added
✔️ Tests pass
✔️ App runs as expected

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**